lib/acts_as_tenant/model_extensions.rb

```diff
@@ -71,15 +71,19 @@ def acts_as_tenant(tenant = :account, **options)
         to_include = Module.new {
           define_method "#{fkey}=" do |integer|
             write_attribute(fkey.to_s, integer)
-            raise ActsAsTenant::Errors::TenantIsImmutable if send("#{fkey}_changed?") && persisted? && !send("#{fkey}_was").nil?
+            raise ActsAsTenant::Errors::TenantIsImmutable if tenant_modified?
             integer
           end
 
           define_method "#{ActsAsTenant.tenant_klass}=" do |model|
             super(model)
-            raise ActsAsTenant::Errors::TenantIsImmutable if send("#{fkey}_changed?") && persisted? && !send("#{fkey}_was").nil?
+            raise ActsAsTenant::Errors::TenantIsImmutable if tenant_modified?
             model
           end
+
+          define_method "tenant_modified?" do
+            send("will_save_change_to_#{fkey}?") && persisted? && !send("#{fkey}_in_database").nil?
```

Comment:
Let's clean this up while we're at it and use the methods from Rails instead of `send`:

```ruby
will_save_change_to_attribute?(fkey) && persisted? && attribute_in_database(fkey).present?
```

https://github.com/rails/rails/blob/main/activerecord/lib/active_record/attribute_methods/dirty.rb#L128